nix/update-dependencies.sh

```diff
@@ -4,7 +4,7 @@
 set -euo pipefail
 
 # lorri should always build with the current NixOS stable branch.
-channel='nixos-19.09'
```

Comment:
Currently, lorri "officially" supports 19.09 - see https://github.com/target/lorri/blob/master/MAINTAINERS.md#publishing-a-release-on-nixpkgs. Switching CI to 20.03 means that we pull in a newer version of Rust. This also means that we may start using Rust features that are not supported by the Rust version shipped with 19.09.

As such, this change is more than just a fix to CI; it implies soft-deprecating 19.09. If there is a way to separate out fixing CI from deprecating 19.09, then I would be in favour of splitting the PR and then having a separate discussion about which NixOS versions we want to support.

Reply:
The main reason I soft-deprecated `nixos-19.09` is because it would consume a huge chunk of time just building the overlayed `nixpkgs-fmt` (easily 10 minutes just to enter the shell for the first time). One way this could reasonably be solved is by maybe utilizing the [`nixpkgs-mozilla` overlay](https://github.com/mozilla/nixpkgs-mozilla/) to pin the Rust packages to 1.37.0 (the version in `nixos-19.09`) and having jobs for both 1.41.0 and 1.37.0 utilizing that overlay. What do you think about that?

Reply:
@cole-h - thanks for suggesting an alternative with the Rust overlay! I believe that what you're saying would work. It would also somewhat complicate the setup.

My main point was mostly that the decision to "effectively" deprecate 19.09 should be taken consciously. I am personally quite happy to drop support for 19.09 now that 20.03 is out, which would allow us to go ahead without the Rust overlay.

@grahamc @Profpatsch @nyarly, are you okay with us dropping support for 19.09 and move to 20.03 as the officially supported version (by way of running CI using 20.03)?

If yes, let's change all occurrences of "19.09" to "20.03" in https://github.com/target/lorri/blob/master/MAINTAINERS.md too (and also in the README in case we mention 19.09 somewhere there).

Reply:
Proposed docs change: https://github.com/target/lorri/pull/399

Reply:
> @grahamc @Profpatsch @nyarly, are you okay with us dropping support for 19.09 and move to 20.03 as the officially supported version (by way of running CI using 20.03)?

Let’s keep support for 19.09 around until 20.09 is released. We can use the pinned mozilla rustpkgs overlay.

Reply:
I've filed https://github.com/target/lorri/issues/406.